cpp/src/arrow/builder.h

```diff
@@ -1055,10 +1066,14 @@ class ARROW_EXPORT StructBuilder : public ArrayBuilder {
 
 namespace internal {
 
-// TODO(ARROW-1176): Use Tensorflow's StringPiece instead of this here.
+// TODO deprecate this
```

Comment:
Any reason it cannot be removed (since this was non-public)?

Reply:
Cool. I'll remove then merge this, as it would be helpful to use in the buffered reader implementation (`Peek` can return string_view instead of a Buffer)